### PR TITLE
📦 [Dependencies.swift][SPM] Fix header mapping

### PR DESCRIPTION
**Pull request by danyf90**
### Short description 📝

Fixes the problems reported in these 2 issues:
- https://github.com/tuist/tuist/issues/3172
- https://github.com/tuist/tuist/issues/3171

Other problems have surfaced regarding other dependencies, and I'll open separate issues and PRs for them 🤞 

### Checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase.
- [ ] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [ ] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [ ] In case the PR introduces changes that affect users, the documentation has been updated.

### CODE REVIEW

File: Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift [hunk 1]
```diff
@@ -358,11 +358,24 @@ extension ResourceFileElements {
...
+        let projectHeaders = allHeaders
+            .filter { !publicHeaders.contains($0) }
+            .filter { !privateHeaders.contains($0) }
+
+        return .init(
+            public: publicHeaders.isEmpty ? nil : .init(globs: publicHeaders.map { Path($0.pathString) }),
+            private: privateHeaders.isEmpty ? nil : .init(globs: privateHeaders.map { Path($0.pathString) }),
+            project: projectHeaders.isEmpty ? nil : .init(globs: projectHeaders.map { Path($0.pathString) })
+        )
     }
```
Comment (danyf90) [inline Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift:379]: I still have to check what is the correct logic here. If anyone knows it, any input is very welcome! 🙌

File: Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift [hunk 2]
```diff
@@ -359,10 +359,16 @@ extension ResourceFileElements {
...
+        let publicHeadersAbsolutePath = path.appending(RelativePath(publicHeadersPath ?? "include"))
+        let publicHeadersPathContent = FileHandler.shared.filesAndDirectoriesContained(in: publicHeadersAbsolutePath, shallow: true)
+        guard
+            let publicHeaders = publicHeadersPathContent?.filter({ $0.extension == "h" }),
+            !publicHeaders.isEmpty
+        else {
+            return nil
+        }
+
+        return .init(public: .init(globs: publicHeaders.map { Path($0.pathString) }))
```
Comment (fortmarek) [inline Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift:371]: I'd probably make the init of at least headers explicit

File: Sources/TuistSupport/Utils/FileHandler.swift [hunk 3]
```diff
@@ -73,7 +73,7 @@ public protocol FileHandling: AnyObject {
     func changeExtension(path: AbsolutePath, to newExtension: String) throws -> AbsolutePath
     func resolveSymlinks(_ path: AbsolutePath) -> AbsolutePath
     func fileAttributes(at path: AbsolutePath) throws -> [FileAttributeKey: Any]
-    func filesAndDirectoriesContained(in path: AbsolutePath) -> [AbsolutePath]?
+    func filesAndDirectoriesContained(in path: AbsolutePath, shallow: Bool) -> [AbsolutePath]?
```
Comment (fortmarek) [inline Sources/TuistSupport/Utils/FileHandler.swift:76]: how is `shallow` different from `contentsOfDirectory`? I'd suggest using that function and removing this parameter

**Review by fortmarek — COMMENTED** [review]
Looks good! I have just two minor comments related to the implementation

**Pull request merged**